src/IO/H5/Helpers.hpp

```diff
@@ -113,9 +113,43 @@ bool contains_attribute(hid_t file_id, const std::string& group_name,
 
 /*!
  * \ingroup HDF5Group
- * \brief Read a DataVector from a dataset in a group
+ * \brief Open a dataset in an HDF5 group with name "dataset_name"
  */
-DataVector read_data(hid_t group_id, const std::string& dataset_name);
+
+hid_t open_dataset(const hid_t group_id,
+                   const std::string& dataset_name) noexcept;
+
+/*!
+ * \ingroup HDF5Group
+ * \brief Use a dataset to open an HDF5 dataspace
+ */
+
+hid_t open_dataspace(const hid_t dataset_id) noexcept;
+
+/*!
+ * \ingroup HDF5Group
+ * \brief Read a scalar dataset into a bool, double, or size_t
+ */
+
+template <typename T>
+T read_scalar_dataset(const hid_t group_id,
+                      const std::string& dataset_name) noexcept;
+
+/*!
+ * \ingroup HDF5Group
+ * \brief Read a rank-1 dataset into a DataVector or std::vector
+ */
+
+template <typename T>
+T read_rank1_dataset(hid_t group_id, const std::string& dataset_name) noexcept;
```

Comment:
What would need to happen to be able to declare one function that takes two template parameters, a `size_t` that is the `Rank` and a second one `T` that is the type?